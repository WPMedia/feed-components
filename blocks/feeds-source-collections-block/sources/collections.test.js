// eslint-disable-next-line no-unused-vars
import Consumer from 'fusion:consumer'
const resolver = require('./collections')

it('validate params', () => {
  expect(resolver.default.params).toStrictEqual({
    _id: 'text',
    content_alias: 'text',
    from: 'text',
    size: 'text',
  })
})

it('returns query search by _id', () => {
  const query = resolver.default.resolve({
    _id: 'NBXKZJC3ONEGRB2VPINBMQTNFA',
    content_alias: '',
    from: '',
    size: '',
  })
  expect(query).toMatchSnapshot()
})

it('returns query by alias', () => {
  const query = resolver.default.resolve({
    _id: '',
    content_alias: 'sports page',
    from: '5',
    size: '10',
  })
  expect(query).toMatchSnapshot()
})
