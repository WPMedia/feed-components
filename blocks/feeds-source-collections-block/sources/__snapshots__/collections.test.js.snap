// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns query by alias 1`] = `"content/v4/collections?content_alias=sports page&from=5&size=10&published=true"`;

exports[`returns query search by _id 1`] = `"content/v4/collections?_id=NBXKZJC3ONEGRB2VPINBMQTNFA&published=true"`;
